### PR TITLE
[#126391] Flag ongoing reservations as problem orders when taking an instrument offline

### PR DESCRIPTION
**Pull request by drench**
This replaces #695.

### CODE REVIEW

File: app/models/order_detail.rb [hunk 1]
```diff
@@ -465,6 +465,10 @@ def complete!
     change_status!(OrderStatus.complete_status)
   end
 
+  def force_complete!
```
Comment (drench) [inline app/models/order_detail.rb]: I considered `#to_problem!` though I thought this name was more direct. This is used to get ongoing reservations flagged as "problem" though it's done indirectly.

File: app/models/order_detail.rb [hunk 2]
```diff
@@ -465,6 +465,10 @@ def complete!
...
+    update(state: "complete", order_status: OrderStatus.complete.first)
```
Comment (jhanggi) [inline app/models/order_detail.rb]: We have `OrderStatus.complete_status` which wraps up the `complete.first`.

File: app/controllers/offline_reservations_controller.rb [hunk 3]
```diff
@@ -52,6 +53,12 @@ def update
 
   private
 
+  def flag_ongoing_reservations_as_problem
+    OrderDetail
+      .where(id: @instrument.reservations.ongoing.pluck(:order_detail_id))
```
Comment (jhanggi) [inline app/controllers/offline_reservations_controller.rb]: I'm not sure a callback is the right answer, but this seems a little too knowledgable about order details for a controller. Not sure I have a better solution, though.

**Comment by jhanggi** [comment]
👍

**Pull request merged**